File: 125-ValidPalindrome/125-ValidPalindrome.cpp
// Last updated: 9/8/2026, 12:36:56 am
class Solution {
public:
    bool isPalindrome(string s) {
        if(s.length() < 2) return true;
        string filtered;
        for (char c : s) {
            if (isalnum(c)) {
                filtered += tolower(c);
            }
        }

        int left = 0;
        int right = filtered.size() - 1;

        while (left < right) {
            if (filtered[left] != filtered[right]) {
                return false;
            }
            left++;
            right--;
        }

        return true;        
    }
};

File: 125-ValidPalindrome/125-ValidPalindrome.py
# Last updated: 9/8/2026, 12:37:00 am
class Solution(object):
    def isPalindrome(self, s):
        d1={}
        ns=""
        for l in s:
            if l.isalnum():
                ns+=l
        ns=ns.lower()
        nsr=""
        for l in ns:
            nsr=l+nsr
        if nsr == ns:
            return True
        else:
            return False
        